File: lib/behavioral/strategy/services/interest_calculator_service.dart
import 'package:gof_design_patterns/behavioral/strategy/models/order.dart';
import 'package:gof_design_patterns/behavioral/strategy/strategies/interest_calculator_strategy.dart';

class InterestCalculatorService {
  InterestCalculatorService({this.interestCalculator});

  InterestCalculatorStrategy interestCalculator;

  double run(Order order) {
    return interestCalculator.getInterestRate(order);
  }
}

File: behavioral/strategy/interestCalculatorService.go
package main

type interestCalculatorService struct {
	InterestCalculatorStrategy iInterestCalculatorStrategy
}

func newInterestCalculatorService(i iInterestCalculatorStrategy) *interestCalculatorService {
	return &interestCalculatorService{
		InterestCalculatorStrategy: i,
	}
}

func (i *interestCalculatorService) run(o order) float64 {
	return i.InterestCalculatorStrategy.getInterestRate(o)
}
